resharper/src/resharper-unity/CSharp/Daemon/Stages/Analysis/InstantiateWithoutParentProblemAnalyzer.cs

```diff
@@ -0,0 +1,169 @@
+            
+            var info = reference.Resolve();
+            if (info.ResolveErrorType != ResolveErrorType.OK) 
+                return;
+
+            var method = info.DeclaredElement as IMethod;
+            if (method == null)
+                return;
+            
+            if (!method.ShortName.Equals(ourKnownMethod)) 
+                return;
+
+            if (method.GetContainingType()?.GetClrName().Equals(KnownTypes.Object) != true)
+                return;
+
+           
+            var parameters = method.Parameters;
+            if (parameters.Count != 1) 
+                return;
+
+            var scope = GetScope(expression);
+            if (scope == null)
+                return;
+            
+            IEnumerable<ITreeNode> usages = null;
+            var containingParenthesizedExpression = expression.GetContainingParenthesizedExpression();
+            var castExpression = CastExpressionNavigator.GetByOp(containingParenthesizedExpression);
+            var asExpression = AsExpressionNavigator.GetByOperand(containingParenthesizedExpression);
+            var initializer = ExpressionInitializerNavigator.GetByValue(castExpression ?? asExpression ?? containingParenthesizedExpression);
+            var declaration = LocalVariableDeclarationNavigator.GetByInitial(initializer);
+
+
+            var usageProvider = data.GetUsagesProvider();
+            if (declaration != null)
+            {
+                usages = usageProvider.GetUsages(declaration.DeclaredElement, scope);
+            }
+            else
+            {
+                var assignment = AssignmentExpressionNavigator.GetBySource(castExpression ?? asExpression ?? containingParenthesizedExpression);
+                var dest = assignment?.Dest as IReferenceExpression;
+                var destInfo = dest?.Reference.Resolve();
+                if (destInfo != null && destInfo.ResolveErrorType == ResolveErrorType.OK)
+                {
+                    usages = usageProvider.GetUsages(destInfo.DeclaredElement.NotNull(), scope);
+                }
+                else
+                {
+                    return;
+                }
+            }
+
+            foreach (var usage in usages)
+            {
+                if (usage is IReferenceExpression referenceExpression)
+                {
+                    var fullReferenceExpression = ReferenceExpressionNavigator.GetTopByQualifierExpression(referenceExpression);
+                    if (IsUsageSetTransformParent(fullReferenceExpression, out var stayInWorldCoords, out var transform))
+                    {
+                        var finder = new TransformParentRelatedReferenceFinder(referenceExpression);
+                        var relatedExpressions = finder.GetRelatedExpressions(scope, expression).FirstOrDefault();
+                        if (relatedExpressions == null || relatedExpressions.GetTreeStartOffset() >= fullReferenceExpression.GetTreeStartOffset())
```

Comment:
I'd also recomment checking that the `SetParent` method is called in the same block. Current code seems to allow cases like this:
```cs
var variable = Instantiate();
if (something)
  variable.SetParent(firstTransform);
else
  variable.SetParent(secondTransform);
```

please add tests for such cases